docs/reference/contributing/storage/storage.md

```diff
@@ -1,3 +1,27 @@
 <h2 id="contributing-storage">Storage</h2>
 
-The two storage APIs are BlockDevice and FileSystem. These APIs provide filesystem and file operations on a block device, as well as an interface for access to block-based storage.
+Storage support is split between file systems and their underlying block device support. The [storage API page](/docs/v5.6/reference/storage-overview.html) has more information on existing APIs in Mbed OS for both interfaces.
+
+#### Block Device
+
+Backing file systems on new hardware requires adding a block device implementation. You can extend the [BlockDevice](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/bd/HeapBlockDevice.h) class to provide support for unsupported storage.
+
+If you want to port a new file system to Mbed OS on existing storage options, you can skip to the following section.
+
+#### File systems
+
+To implement a new file system in Mbed OS, an implementor needs to provide the abstract functions in the file system interface. The [FAT file system](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/fat/FATFileSystem.cpp) provides an excellent example. You can find tests of the POSIX API here.
+
+A minimal file system needs to provide the following functions:
+
+- `file_open`.
+- `file_close`.
+- `file_read`.
+- `file_write`.
+- `file_seek`.
+
+Here is the full API that a file system may implement:
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://github.com/ARMmbed/mbed-os/blob/master/features/filesystem/FileSystem.h#L205)
+
+A block device must back file systems in Mbed OS. If you are using supported hardware, then you can continue. Otherwise, view the block device porting section above.
```

Comment:
Query: What does continue mean here? Read the next page or continue what the reader's doing? Please clarify.

Reply:
This was worded a bit poorly, continue as in continue with your filesystem porting. Otherwise you'll need a block device port to back the filesystem and should implement that first.